gc/gc-base/src/test/java/org/projectnessie/gc/serialization/ReferencesSerializerTest.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.gc.serialization;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import java.io.ByteArrayOutputStream;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.zip.DeflaterOutputStream;
+import org.junit.jupiter.api.Test;
+import org.projectnessie.gc.base.ContentBloomFilter;
+import org.projectnessie.gc.base.ImmutableGCParams;
+import org.projectnessie.model.ImmutableBranch;
+import org.projectnessie.model.ImmutableCommitMeta;
+import org.projectnessie.model.ImmutableReferenceMetadata;
+import org.projectnessie.model.ImmutableTag;
+import org.projectnessie.model.Reference;
+
+class ReferencesSerializerTest {
+
+  @Test
+  public void shouldSerializerAndDeserializerImmutableBranch() {
+    // given
+    Map<String, String> properties = new HashMap<>();
+    properties.put("key", "value");
+    properties.put("key2", "value2");
+
+    Kryo kryo = new Kryo();
+    new ReferencesKryoRegistrator().registerClasses(kryo);
+    ImmutableBranch immutableBranch =
+        ImmutableBranch.builder()
+            .hash("0a62a2d020df785dfb3d34e9fb64d965e6e65220")
```

Comment:
I believe this needs a few more tests where for example everything that's `Nullable` in `Branch` is actually omitted when creating a `Branch` instance.

Given that, I think it would be good if there would be a separate test class for each custom serializer and within that test class we can then make sure all edge cases are covered

Reply:
agree, I wanted to wait until we will settle out whether we want to use the kryo or not. If we decide that this is a good direction, I will add more unit tests.